haskell/src/Monocle/Main.hs

```diff
@@ -35,7 +35,7 @@ type MonocleAPI' = MonocleAPI :<|> AuthAPI
 type RootAPI = "api" :> "2" :> MonocleAPI' :<|> MonocleAPI'
 
 type AuthAPI =
-  "auth" :> "login" :> Get '[JSON] NoContent
+  "auth" :> "login" :> QueryParam "uri" Text :> Get '[JSON] NoContent
```

Comment:
To avoid uri encoding issue, we can also use [`CaptureAll`](https://hackage.haskell.org/package/servant-0.19/docs/Servant-API.html#t:CaptureAll)

Reply:
I don't think CaptureAll fills well the need as it capture path segments however it seems a simple parameter in the query string is well adapted to the need.